src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -203,12 +262,19 @@ private static ServiceClientParameter BuildParameter(Parameter requestParameter)
                 defaultValue = ClientModelBuilderHelpers.ParseClientConstant(constantSchema);
             }
 
+            ParameterLocation? location = null;
```

Comment:
rem

Reply:
I can't remove Location logic or else BuildNextPageMethod won't work. It doesn't have access to the Parameter to get Parameter.Protocol.Http to make the decision on how to filter the parameters. So, it needs to be baked into ServiceClientParameter.

Reply:
We can figure out correct parameters by looking at original method header values.

Reply:
Fixed.